lib/iris/fileformats/grib/_load_convert.py

```diff
@@ -758,6 +758,59 @@ def check_range(v1, v2, n, d):
     metadata['dim_coords_and_dims'].append((x_coord, x_dim))
 
 
+def grid_definition_template_40(section, metadata):
+    """
+    Translate template representing a regular Gaussian grid.
+
+    Updates the metadata in-place with the translations.
+
+    Args:
+
+    * section:
+        Dictionary of coded key/value pairs from section 3 of the message.
+
+    * metadata:
+        :class:`collections.OrderedDict` of metadata.
+
+    """
+    scan = scanning_mode(section['scanningMode'])
+
+    major, minor, radius = ellipsoid_geometry(section)
+    cs = ellipsoid(section['shapeOfTheEarth'], major, minor, radius)
+
+    # Calculate longitude points.
+    x_inc = section['iDirectionIncrement'] * _GRID_ACCURACY_IN_DEGREES
+    x_offset = section['longitudeOfFirstGridPoint'] * _GRID_ACCURACY_IN_DEGREES
+    x_direction = -1 if scan.i_negative else 1
+    Ni = section['Ni']
+    x_points = np.arange(Ni, dtype=np.float64) * x_inc * x_direction + x_offset
+
+    # Determine whether the x-points (in degrees) are circular.
+    circular = _is_circular(x_points, 360.0)
+
+    # Get the latitude points.
+    y_points = section['distinctLatitudes']
+    y_points.sort()
```

Comment:
It was inherited from the previous implementation. I just did a quick test and it seems this may not be necessary as the `distinctLatitudes` key seems to return values consistent with the `jScansPositively` key. I don't remember how we/I made the original decision to sort and reorder.